sw/device/lib/testing/fault_handler_testutils.c

```diff
@@ -0,0 +1,45 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/csr.h"
+#include "sw/device/lib/runtime/log.h"
+
+static uint32_t get_mepc(void) {
+  uint32_t mepc;
+  CSR_READ(CSR_REG_MEPC, &mepc);
+  return mepc;
+}
+
+static void set_mepc(uint32_t mepc) { CSR_WRITE(CSR_REG_MEPC, mepc); }
+
+void fault_handler_testutils_set_return_addr(uint32_t *indirection) {
+  uint32_t mepc = get_mepc();
+  LOG_INFO("Fault address: mepc = 0x%x", mepc);
+
+  if (indirection == NULL) {
+    // Check if the two least significant bits of the instruction are b11 (0x3),
```

Comment:
This should probably go in some utility somewhere.
```c
bool is_compressed_rv_instruction(uint32_t);
```

I have no strong feelings about where to put it. `memory.h` is probably fine?? @cfrantz

Reply:
I don't think this will be necessary, since very shortly I will be migrating this test to use the OTTF which takes care of this, as I described above. I.e., this test will only need to override the `ottf_instr_access_fault()` ISR, as the main (parent) OTTF exception handler will take care of this (see #9209), allowing this sort fo processing to be opaque to the test writer.

Reply:
Done.